### ISSUE TITLE
Keyboard shortcut for Save

### ISSUE THREAD

**Issue by mnorbeck** [HospitalRun/hospitalrun-frontend#540]
Request is to provide keyboard shortcuts for Save.   Ideally both Control-S and Command-S would be supported.

**Comment by dlo83** [comment]
Does anyone have a problem with using an ember addon like ember-keyboard to accomplish this functionality?  If not, I'd be happy to give this a stab.

**Comment by jglovier** [comment]
>Does anyone have a problem with using an ember addon like ember-keyboard to accomplish this functionality?

@dlo83 I think that could be a great solution. @jkleinsc any objections to that?

**Comment by jkleinsc** [comment]
@jglovier and @dlo83 that is fine by me.

**Comment by dlo83** [comment]
@jglovier, @jkleinsc : sweet, I'll take a crack at this.

**Comment by dlo83** [comment]
FWIW: ember-keyboard doesn't yet support the command modifier key on Macs.  Would a Ctrl+S shortcut suffice for the time being?

**Comment by jkleinsc** [comment]
@dlo83 wouldn't the command modifier be "OSRight" and "OSLeft"?  I took a quick look at ember-keyboard and it appears this would work, but I didn't test it, so maybe there is a bug that I don't know about?

**Comment by dlo83** [comment]
@jkleinsc one would think.  However, I tried those command modifiers and was unable to trigger the update action with them. Ctrl+S worked as expected.  The ember-keyboard repo explicitly states: Support for keyup, keydown, and keypress, as well as the modifier keys: ctrl, alt, shift, and meta.  Perhaps the omission of command is intentional.  I'll follow up with the developer of that repo.

**Comment by dlo83** [comment]
@mnorbeck I think Ctrl+S/Cmd+S might be a problem here.  It appears as though some browsers like Chrome don't allow you to override common keyboard shortcuts that are already registered in the browser (Ctrl+S, Ctrl+W, Ctrl+C, etc).  Would you consider any alternatives? Perhaps adding an additional modifier key (Ctrl+Alt+S)?

**Comment by amyrlam** [comment]
I'll do this one with @alychloe

**Comment by amyrlam** [comment]
👋 Let's reopen this issue for tracking, I still need to add the keyboard shortcut to all Update or Save buttons. https://github.com/HospitalRun/hospitalrun-frontend/pull/665 implements the keyboard shortcut in one place.

**Comment by taras** [comment]
@amyrlam can are you able to make a list of all of the places where this change needs to be applied? Also, we have some mixins that are used across all edit screens. Could we add this functionality as a mixin to one of those mixins?

**Comment by amyrlam** [comment]
@taras sure, sounds good, I'll take a look at that and reach out if questions

**Comment by jscottchapman** [comment]
@taras @amyrlam  It looks like this one is in progress/being worked on.  Is that right?  Just wanted to make sure that someone didn't start working on the same issue unknowingly.

**Comment by amyrlam** [comment]
@jscottchapman I'm working on this one! I'll try to report back soon

**Comment by jkleinsc** [comment]
@amyrlam thanks for the update.  I'm going to mark this item as in progress then.

**Comment by tangollama** [comment]
@amyrlam are you still interested in this issue? If not, I'll move it back to Help Wanted.

**Comment by amyrlam** [comment]
move it to help wanted, but i try to work on some hospital run again soon!

**Comment by tangollama** [comment]
Thx for the reply @amyrlam. Now worries. We're all volunteering. :)

**Comment by DerekTBrown** [comment]
@tangollama would love to take on this as well.

**Comment by tangollama** [comment]
Have at it. Thx.
